src/Features/Core/Portable/Diagnostics/HostAnalyzerManager.cs

```diff
@@ -171,14 +172,12 @@ public bool IsAnalyzerSuppressed(DiagnosticAnalyzer analyzer, Project project)
                 return true;
             }
 
-            // don't capture project
-            var projectId = project.Id;
-
-            // Skip telemetry logging for supported diagnostics, as that can cause an infinite loop.
-            void onAnalyzerException(Exception ex, DiagnosticAnalyzer a, Diagnostic diagnostic) =>
-                    AnalyzerHelper.OnAnalyzerException_NoTelemetryLogging(a, diagnostic, _hostDiagnosticUpdateSource, projectId);
-
-            return CompilationWithAnalyzers.IsDiagnosticAnalyzerSuppressed(analyzer, options, onAnalyzerException);
+            // NOTE: Previously we used to return "CompilationWithAnalyzers.IsDiagnosticAnalyzerSuppressed(options)"
+            //       on this code path, which returns true if analyzer is suppressed through compilation options.
+            //       However, this check is no longer correct as analyzers can be enabled/disabled for individual
+            //       documents through .editorconfig files. So we pessimistically assume analyzer is not suppressed
+            //       and let the core analyzer driver in the compiler layer handle skipping redundant analysis callbacks.
```

Comment:
and just to validate, but your test ensures that the analyzer *is* skipped in this scenario?

Reply:
Test that I added validates that no diagnostic was reported. We have existing tests in compiler layer that validate that disabled analyzers are skipped by analyzer driver.

Reply:
ok thanks :)